src/main/java/core/training/TrainingManager.java

```diff
@@ -23,12 +24,13 @@
 	// singleton class
 	private static TrainingManager m_clInstance;
 
-	private TrainingPerWeek nextWeekTraining;        // used to determine training bar, include upcoming game   => Created at initilization
-    private TrainingWeekManager recentTrainings;     // trainings that took place (if any null otherwise) since last entry in Training table  => Created at initilization
-	private List<TrainingPerWeek> historicalTrainings;         // used to populate training history, no match information => Created at initilization
+	private TrainingPerWeek nextWeekTraining;        						// used to determine training bar, include upcoming game   => Created at initilization
+    private TrainingWeekManager recentTrainings;     						// trainings that took place (if any null otherwise) since last entry in Training table  => Created at initilization
+	private List<TrainingPerWeek> historicalTrainings;          			// used to populate training history, no match information => Created at initilization
+	private List<TrainingPerWeek> historicalTrainingsInclMatches;           // used for subskill recalculation => Created on request
```

Comment:
i do not see any difference between both lists. Both are including match info and they are identical if computeHistoricalTrainingsInclMatches uses the same start date as the TrainingManager.instance(). so i propose:

* use historicalTrainings to filter in getTrainingsBetween
* remove the loading of matches from TrainingPerWeek constructor and make getMatches loading data on request again

getMatches(){
if (matches==null) loadmatches();
return matches;
}

Reply:
> i do not see any difference between both lists. Both are including match info

no, first list is simply the content of training table loaded into a List there are no match info .... it is very quick to load. 
the second one is made parsing HRFs, and loading all matchkurzinfo (more heavy)

we only want to load match for subskill calculation, otherwise we don't even need a training table at all  and we redo the work of parsing hrf with match info at every application start (but I believe this would be terrible on performance).

Reply:
loading the training table content invokes the constructor of TrainingPerWeek which loads the matches - isn't it?

Reply:
no it does not ...

Reply:
Let's try with this PR, and see how it goes, ok for you ?